### PR TITLE
Feature/posix

### PR DESCRIPTION
**Pull request by greymd**
Changes
* Issue #9: Be compatible with POSIX **as much as possible**.
  + abolish `xmllint` and `which`
  + `curl` and `perl` are still remaining because they are general enough.
* Add some test cases for negative testing.
* some refactoring

完全POSIX準拠じゃないけど、まあこんくらいでいいっしょってことで。

### CODE REVIEW

File: wpb.sh [hunk 1]
```diff
@@ -67,12 +67,14 @@ is_env_ok () {
         type $cmd > /dev/null 2>&1
         if [ $? -ne 0 ]; then
             echo "$cmd is required to work." >&2
-            return -1
+            # `return -1` does not work in particular situation. `-1` is recognized as an option.
```
Comment (mecab) [inline wpb.sh:70]: 😱

**Review by mecab — APPROVED** [review]
[LGTM](https://lgtm.in/i/gcAQ04qcq)
